File: src/main/java/com/TIComoApp/TIComo/model/Plato.java
package com.TIComoApp.TIComo.model;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import com.mongodb.lang.NonNull;

import lombok.Data;

@Data
@Document(collection = "platos")
public class Plato {

	@Id
	private String id;
	@NonNull
	private String nombre;
	@NonNull
	private String foto;
	@NonNull
	private String descripcion;
	@NonNull
	private double precio;
	@NonNull
	private boolean aptoVeganos;
	@NonNull
	private String nombreRestaurante;
	
	public Plato(String id, String nombre, String foto, String descripcion, double precio, boolean aptoVeganos,
			String nombreRestaurante) {
		this.id = id;
		this.nombre = nombre;
		this.foto = foto;
		this.descripcion = descripcion;
		this.precio = precio;
		this.aptoVeganos = aptoVeganos;
		this.nombreRestaurante = nombreRestaurante;
	}

	public String getId() {
		return id;
	}

	public void setId(String id) {
		this.id = id;
	}

	public String getNombre() {
		return nombre;
	}

	public void setNombre(String nombre) {
		this.nombre = nombre;
	}

	public String getFoto() {
		return foto;
	}

	public void setFoto(String foto) {
		this.foto = foto;
	}

	public String getDescripcion() {
		return descripcion;
	}

	public void setDescripcion(String descripcion) {
		this.descripcion = descripcion;
	}

	public double getPrecio() {
		return precio;
	}

	public void setPrecio(double precio) {
		this.precio = precio;
	}

	public boolean isAptoVeganos() {
		return aptoVeganos;
	}

	public void setAptoVeganos(boolean aptoVeganos) {
		this.aptoVeganos = aptoVeganos;
	}

	public String getNombreRestaurante() {
		return nombreRestaurante;
	}

	public void setNombreRestaurante(String nombreRestaurante) {
		this.nombreRestaurante = nombreRestaurante;
	}

	@Override
	public String toString() {
		return "Plato [id=" + id + ", nombre=" + nombre + ", foto=" + foto + ", descripcio=" + descripcion + ", precio="
				+ precio + ", aptoVeganos=" + aptoVeganos + ", idRestaurante=" + nombreRestaurante + "]";
	}

	
	
}

File: src/main/webapp/cliente/src/app/models/Plato.ts
export class Plato {
	/*private*/ id?: String;
    /*private*/ nombre :String;
    /*private*/ foto: String;
    /*private*/ descripcion: String;
    /*private*/ precio: Number;
    /*private*/ aptoVeganos: boolean;
    /*private*/ nombreRestaurante: String;
    
    
    constructor(Nombre: String, Foto: String, Descripcion: String,Precio: Number,AptoVeganos: boolean, nombreRestaurante:String){
        this.nombre = Nombre;
        this.foto = Foto;
        this.descripcion = Descripcion;
        this.precio = Precio;
        this.aptoVeganos = AptoVeganos;
		this.nombreRestaurante= nombreRestaurante;
        
    }
	get_id(): String | undefined {
		return this.id;
	}

	getNombre(): String {
		return this.nombre;
	}

	getFoto(): String {
		return this.foto;
	}
	getDescripcion(): String {
		return this.descripcion;
	}
	getPrecio(): Number {
		return this.precio;
	}
	getAptoVeganos(): boolean {
		return this.aptoVeganos;
	}
	getnombreRestaurante(): String {
		return this.nombreRestaurante;
	}
}